### DISCUSSION TITLE
What do we mean by reusing software?

### DISCUSSION THREAD

**Discussion by c-martinez** [NLeSC/guide#237] [Ideas]
Some nice links of different options for reuse. I think it would be useful to link it in the guide.

http://sites.computer.org/debull/A18mar/A18MAR-CD.pdf#page=17

https://drops.dagstuhl.de/opus/volltexte/2016/5817/pdf/dagrep_v006_i001_p108_s16041.pdf#page=23

**Comment by egpbos** [comment]
I like it! This PRIMAD model seems very useful to specify exactly what we mean and hence to better and more clearly determine our priorities and describe our achievements in projects.

**Comment by c-martinez** [comment]
Yes, I agree. The Turing Way already include a [table of definitions](https://the-turing-way.netlify.app/reproducible-research/overview/overview-definitions.html) for reproducible and replicable, which is not quite as detailed as the links above. I'll see if I can convince them of including extendable and modifiable in their table.

See https://github.com/alan-turing-institute/the-turing-way/issues/1596

**Comment by LourensVeen** [comment]
Those models in the links seem to be limited to computational models and data processing scripts. That's mostly our scope, but for the occasional mishap where we end up developing a library or other such generic component, we may need something else as well.

**Comment by c-martinez** [comment]
Possibly. I feel that: "Modifiable. We say an experiment is modifiable if we can change its implementation for reuse purposes, and this is achievable by having the source code of the experiment." could apply to a library. In any case, would be good to have your input when we start working on extending this chapter in the Turing Way.

**Comment by LourensVeen** [comment]
Hmm, in my opinion thinking of libraries as chunks of code that you download and modify to suit your needs is a recipe for disaster, but I'm an engineer. I've certainly met scientists who thought that it was a good idea, so maybe it should be in the Guide or Way.

**Comment by bouweandela** [comment]
I converted the issue to a discussion, to try out if we like the discussions format better for the guide than the issue format. I hope that's not inconvenient for anyone!
